### ISSUE TITLE
Dot support for graph-of-graphs

### ISSUE THREAD

**Issue by jakevossen5** [petgraph/petgraph#417]
### Summary

I have a data structure that is a graph of graphs. I would like to be able to visualize it all in one go in the `.dot` format

### Motivation

Summary gets the main point. It would be totally okay if the main graph just point big nodes, where in the nodes custom `.dot` code could be outputted. I don't think it makes sense for it to be connected to a particular start node.

### Details

Maybe a new trait could be added that is called when generating `.dot` files, and you can implement it to change the default behavior of just taking the debug information out of the struct? I currently don't have the bandwidth to work on this and also realize it is probably a pretty fringe feature request.

**Comment by saona-raimundo** [comment]
Hi @jakevossen5!

I am no maintainer, but I am curious, you are using a graph of graphs.... do you actually connect any node from one graph to another? 

## Side note
I have a [WIP crate](https://github.com/saona-raimundo/ferret_hypergraph) (not yet published to crates.io) that bornt almost only from this need. 
I already implemented a dot conversion and, to be fair, it was not that hard. 
So, I want to understand your use-case better before I try to do something for petgraph.

**Comment by jakevossen5** [comment]
No, no individual node of one graph is connected to a specific node to the other graph. Instead, each graph is now a node that is part of a larger graph.

**Comment by saona-raimundo** [comment]
To make an implementation with minimal impact in the current API, one option is to expose a way to 

1) Choose the [`node_id`](https://graphviz.org/doc/info/lang.html) of a node in the dot format. 
This is currently [implemented here](https://docs.rs/petgraph/latest/src/petgraph/dot.rs.html#163) and uses [`to_index`](https://docs.rs/petgraph/latest/petgraph/visit/trait.NodeIndexable.html#tymethod.to_index)

2) Add a name for the graph 
This is [implemented here](https://docs.rs/petgraph/latest/src/petgraph/dot.rs.html#158) and, currently, the dot graphs have no name.

3) Allow the subgraph type
This is [implemented here](https://docs.rs/petgraph/latest/src/petgraph/dot.rs.html#60) and there is only `graph` and `digraph` types.

This would allow to use the current API, in the following way:

1) Write the graph with [`Config::GraphContentOnly`](https://docs.rs/petgraph/latest/petgraph/dot/enum.Config.html)
... but with custom id for the nodes adding the prefix [`cluster`](https://graphviz.org/doc/info/lang.html#subgraphs-and-clusters).
This would write all nodes (which in this case are graphs) and the edges between them.

2) Write the dot of each node (which are graphs) with the type `subgraph` (instead of graph or digraph)

3) Put all this together in the same `String` (by hand I would say), enclosed in-between `(di)graph {` and `}`.

I would do a PR with these changes... I started the discussion on the first change (choose the `node_id`) in #194, which seems the most tricky part in my opinion.

**Comment by saona-raimundo** [comment]
Just to give some other options (I know this is a feature request to the library)

The trait approach is implemented in [`rustc-ap-graphviz`](https://crates.io/crates/rustc-ap-graphviz) with
the "intention of this library is to emit a human-readable .dot file with very regular structure suitable for easy post-processing."

With this library, I am sure you can do the graph you are looking for.

Other libraries to generate `.dot` files are:
- [`dot`](https://crates.io/crates/dot) - Also takes the trait-implementation route.
- [`dot-writer`](https://crates.io/crates/dot-writer) - This one has explicit [`subgraph`](https://docs.rs/dot-writer/0.1.2/dot_writer/struct.Scope.html#method.subgraph) support, so it should be easy to do what you want.
